python/ray/tests/test_basic_2.py

```diff
@@ -720,6 +720,29 @@ def get(self):
         assert b"OK" in output
 
 
+def test_use_dynamic_function_and_class(ray_start_regular_shared):
+    # Test use dynamically defined functions
+    # and classes for remote tasks and actors.
+    # See https://github.com/ray-project/ray/issues/12834.
+
+    current_path = os.path.dirname(__file__)
+    job_config = ray.job_config.JobConfig(code_search_path=[current_path])
+    ray.init({}, job_config=job_config)
+
+    @ray.remote
+    def foo():
+        return "OK"
+
+    @ray.remote
+    class Foo:
+        def foo(self):
+            return "OK"
+
+    assert ray.get(foo.remote()) == "OK"
+    foo_actor = Foo.remote()
+    assert ray.get(foo_actor.foo.remote()) == "OK"
```

Comment:
This doesn't test if dynamic functions are actually stored in GCS. Can you actually use the Redis client and parse the function table to verify that?